Src/DependencyCollector/Shared.Tests/Implementation/Operation/CacheBasedOperationHolderTests.cs

```diff
@@ -95,6 +96,19 @@ public void GetReturnsItemIfItExistsInTheTable()
             Assert.AreEqual(this.telemetryTuple, this.cacheBasedOperationHolder.Get(id));
         }
 
+        [TestMethod]
+        public void TestItemExpiration()
+        {
+            long id = 911911;
+            this.cacheBasedOperationHolder.Store(id, this.telemetryTuple);
+            Assert.AreEqual(this.telemetryTuple, this.cacheBasedOperationHolder.Get(id));
+
+            // Sleep for 2 secs which is more than the cache expiration time.
+            Thread.Sleep(2000);
```

Comment:
Two comments:
1) It would be good if this test explicitly setup the cache policy it expects (and restores stated when finished if needed).
2) Try to make it faster, a lot faster, sleeping for two seconds in a test is not ideal (when it's one test it doesn't seem so bad but when it is hundreds of tests and they all sleep a bit suddenly the unit tests take a long time to run).